### PR TITLE
Add publisher/server and autoconnect args to BT Monitor

### PR DESCRIPTION
**Pull request by sea-bass**
This PR adds some extra options to Groot to configure the BT monitor to start with specific address and publisher/server ports, as well as automatically connect by "clicking" the Connect button in the code, if enabled.

Specifically, it lets us run Groot as follows:

```
ros2 run groot Groot --mode monitor --address 192.168.1.42 --publisher_port 1668 --server_port 1669 --autoconnect
```

The `autoconnect` option effectively does the following:
- Increases the timeout for the ZMQ subscriber to 10 seconds
- Triggers a click of the "Connect" button in code
- The above resets the timeout to the default 1 second for users to manually connect/disconnect with the normal timeout

This addresses https://github.com/BehaviorTree/Groot/issues/133 and https://github.com/BehaviorTree/Groot/issues/166.

### CODE REVIEW

File: bt_editor/mainwindow.cpp [hunk 1]
```diff
@@ -96,14 +103,20 @@ MainWindow::MainWindow(GraphicMode initial_mode, QWidget *parent) :
...
     ui->leftFrame->layout()->addWidget( _monitor_widget );
 
     connect( ui->toolButtonConnect, &QToolButton::clicked,
             _monitor_widget, &SidepanelMonitor::on_Connect );
 
     connect( _monitor_widget, &SidepanelMonitor::connectionUpdate,
             this, &MainWindow::onConnectionUpdate );
+
+    if (monitor_autoconnect) {
+        usleep(1e6);  // Add a slight delay
```
Comment (sea-bass) [inline bt_editor/mainwindow.cpp:117]: Not super thrilled about this delay, so looking for input here.

File: bt_editor/mainwindow.h [hunk 2]
```diff
@@ -42,7 +42,11 @@ class MainWindow : public QMainWindow
                               SUBTREE_REFRESH};
 
 public:
-    explicit MainWindow(GraphicMode initial_mode, QWidget *parent = nullptr);
+    explicit MainWindow(GraphicMode initial_mode,
+                       const QString& monitor_pub_port = "",
+                       const QString& monitor_srv_port = "",
```
Comment (sea-bass) [inline bt_editor/mainwindow.h:47]: Could also use `std::optional` here if preferred (dunno if we're cool with C++17)

File: bt_editor/sidepanel_monitor.h [hunk 3]
```diff
@@ -15,7 +15,9 @@ class SidepanelMonitor : public QFrame
     Q_OBJECT
 
 public:
-    explicit SidepanelMonitor(QWidget *parent = nullptr);
+    explicit SidepanelMonitor(QWidget *parent = nullptr,
+                              const QString &publisher_port = "",
+                              const QString &server_port = "");
```
Comment (sea-bass) [inline bt_editor/sidepanel_monitor.h:20]: Same here re: `std::optional`

File: bt_editor/mainwindow.cpp (hunk 1 above)
Comment (Timple) [reply bt_editor/mainwindow.cpp:117]: Not sure about alternatives, but the comment can better state why the delay is required. Now the comment is a less brief version of the code.

Comment (sea-bass) [reply bt_editor/mainwindow.cpp:117]: I was thinking maybe no delay, but to add a configurable timeout argument to the "connect" button callback that is more generous during this autoconnect action vs. manually clicking the button.

Comment (sea-bass) [reply bt_editor/mainwindow.cpp]: OK, addressed this!

**Comment by Timple** [comment]
The ports are already optional, could the server ip also be passed? That would be easy to monitor a robot on a dev pc.

**Comment by sea-bass** [comment]
> The ports are already optional, could the server ip also be passed? That would be easy to monitor a robot on a dev pc.

Done!

**Comment by kundezui** [comment]
The following error occurs during Groot compilation, how to solve it?

/home/qwert/BehaviorTree.CPP/Groot/bt_editor/sidepanel_monitor.cpp: In member function ‘void SidepanelMonitor::on_timer()’:
/home/qwert/BehaviorTree.CPP/Groot/bt_editor/sidepanel_monitor.cpp:61:41: error: no matching function for call to ‘zmq::socket_t::recv(zmq::message_t&)’
         while(  _zmq_subscriber.recv(msg) )
                                         ^
In file included from /home/qw/BehaviorTree.CPP/Groot/bt_editor/sidepanel_monitor.h:5:0,
                 from /home/qw/BehaviorTree.CPP/Groot/bt_editor/sidepanel_monitor.cpp:1:
/usr/include/zmq.hpp:625:23: note: candidate: size_t zmq::socket_t::recv(void*, size_t, int)
         inline size_t recv (void *buf_, size_t len_, int flags_ = 0)
                       ^~~~
/usr/include/zmq.hpp:625:23: note:   candidate expects 3 arguments, 1 provided
/usr/include/zmq.hpp:635:21: note: candidate: bool zmq::socket_t::recv(zmq::message_t*, int)
         inline bool recv (message_t *msg_, int flags_ = 0)
                     ^~~~
/usr/include/zmq.hpp:635:21: note:   no known conversion for argument 1 from ‘zmq::message_t’ to ‘zmq::message_t*’
/home/qwert/BehaviorTree.CPP/Groot/bt_editor/sidepanel_monitor.cpp: In member function ‘bool SidepanelMonitor::getTreeFromServer()’:
/home/qwert/BehaviorTree.CPP/Groot/bt_editor/sidepanel_monitor.cpp:149:39: error: ‘zmq::send_flags’ has not been declared
         zmq_client.send(request, zmq::send_flags::none);
                                       ^~~~~~~~~~
/home/qwert/BehaviorTree.CPP/Groot/bt_editor/sidepanel_monitor.cpp:151:60: error: ‘zmq::recv_flags’ has not been declared
         auto bytes_received  = zmq_client.recv(reply, zmq::recv_flags::none);
                                                            ^~~~~~~~~~
CMakeFiles/behavior_tree_editor.dir/build.make:382: recipe for target 'CMakeFiles/behavior_tree_editor.dir/bt_editor/sidepanel_monitor.cpp.o' failed
make[2]: *** [CMakeFiles/behavior_tree_editor.dir/bt_editor/sidepanel_monitor.cpp.o] Error 1
CMakeFiles/Makefile2:109: recipe for target 'CMakeFiles/behavior_tree_editor.dir/all' failed
make[1]: *** [CMakeFiles/behavior_tree_editor.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2

**Pull request merged**